src/Main.elm

```diff
@@ -1,10 +1,132 @@
 module Main exposing (main)
 
-import Data.Audience
-import Data.AudienceFolder
+import Data.Audience exposing ( Audience, subaudiences0 )
+import Data.AudienceFolder exposing ( AudienceFolder, subfolders0, folders0 )
 import Html exposing (Html)
+import Html.Events as Events
+import Html.Attributes as Attr
 
+import Browser
+
+import Dict exposing ( Dict )
+
+-- TODO: remove
+import Debug exposing ( log, todo, toString )
+
+-- ====== Overview ======
+-- 0. To compile the app, execute
+--      $ elm make src/Main.elm --output=public/main.js
+--    and open public/index.html. The css is in public/style.css and is not included when the app is executed via elm reactor.
+-- 1. Data.Audience and Data.AudienceFolder expose dictionaries
+--    whose keys are folder ids and whose values are folders, subfolders, and subaudience of the mentioned folder id.
+-- 2. If all parsing of JSON goes well, the state of the application is just
+--    a triple (record) of parent folder, its subfolders and subaudiences.
+-- 3. The only permissible action is to change the current parent folder.
+
+-- ===STATE=== --
+type alias ExplicitFolder =
+  { parentFolder : AudienceFolder
+  , subfolders : List AudienceFolder
+  , subaudiences : List Audience }
+
+type alias State = Maybe ExplicitFolder
+
+-- a helper function - not worth to put into a separate module
+listOfMaybeList : Maybe (List a) -> List a
+listOfMaybeList mxs =
+  case mxs of
+    Just xs -> xs
+    Nothing -> []
```

Comment:
`listOfMaybeList = Maybe.withDefault []`?

Reply:
Yes, that would have been better.